FunctionApp/Middleware/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace FunctionApp.Middleware
+{
+    public static class ApplicationBuilderExtensions
+    {
+        public static IFunctionsWorkerApplicationBuilder UseSampleMiddleware(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            builder.Services.AddSingleton<SampleMiddleware>();
+
+            builder.Use(next =>
```

Comment:
@fabiocav would this block be necessary if the middleware component, `SampleMiddleware`, implements a specific contract and all implementation of that contract would be resolved by Functions and invoked? That kind of model would make it much simpler as a middleware registration would be literally just registering the components in the container.

Reply:
Absolutely! We're on the same page here :)

This is the base "raw" way the registration will ultimately take place, but our goal is to simplify this process and have a model that simply works with the `Use` extension methods.

We're tracking this work [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/92) and this will likely be the next piece of code I add to the worker.